### PR TITLE
[24.0] Use the new column-select component in `FormData` for multiple select

### LINKED ISSUE #17947: New multiple datasets select should be more efficient with many datasets
**Describe the issue**
The old version of the Multiple datasets select element let you shift- or ctrl-select multiple datasets, which was specially useful for selecting a group of datasets lying close together in a large history.

image

In its new form, one can only select one dataset at a time, which requires the user to scroll down a potentially long dropdown list several times:

image

The filter as you type functionality in the Select a dataset text box helps a bit here:

image

but, unfortunately, the typed filter text gets reset upon every selection.

The best way to populate the select element with many datasets that I'm currently aware of is the new multiselect & drag&drop from the history feature, but I think the tool interface should also offer a more efficient method directly.

One idea would be to offer the new column-select element for options also for Multiple datasets.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 24.0 (not sure whether the interface change happened in 23.2 already)

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
This allows for easier multiselect with added ease for range select. Fixes https://github.com/galaxyproject/galaxy/issues/17947

| Before | Now ➕ |
| ------ | -------- |
| <video src="https://github.com/galaxyproject/galaxy/assets/78516064/722662cb-dda8-41ae-8821-a37f7f9fb2d3" /> | <video src="https://github.com/galaxyproject/galaxy/assets/78516064/b40fa658-81c8-432d-8c93-31d2b34f23f9" />|

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by ahmedhamidawan** [comment]
Should this target 23.2?

**Comment by davelopez** [comment]
Targeting 24.0 sounds good, the multi-select component was added in that version IIRC.

File: client/src/components/Form/Elements/FormSelectMany/FormSelectMany.vue [hunk 1]
```diff
@@ -65,7 +65,7 @@ const searchRegex = computed(() => {
 /** Wraps value prop so it can be set, and always returns an array */
 const selected = computed({
     get() {
-        return Array.isArray(props.value) ? props.value : [props.value];
+        return props.value === null ? [] : Array.isArray(props.value) ? props.value : [props.value];
```
Comment (martenson) [inline client/src/components/Form/Elements/FormSelectMany/FormSelectMany.vue]: this double ternary op is giving me the creeps, maybe just me though :D

Comment (ahmedhamidawan) [reply client/src/components/Form/Elements/FormSelectMany/FormSelectMany.vue]: Ok fixing that

Comment (martenson) [reply client/src/components/Form/Elements/FormSelectMany/FormSelectMany.vue]: thanks @ahmedhamidawan !

**Review by itisAliRH — APPROVED** [review]

**Pull request merged**